src/main.ts

```diff
@@ -616,7 +619,9 @@ function makeDeepPartial(options: Options, longs: ReturnType<typeof makeLongUtil
 
   // Based on the type from ts-essentials
   const keys =
-    options.outputTypeAnnotations || options.outputTypeRegistry ? code`Exclude<keyof T, '$type'>` : code`keyof T`;
+    (options.outputTypeAnnotations || options.outputTypeRegistry) && options.outputTypeAnnotations !== "const-only"
```

Comment:
Hm, this logic is doing in enough places, can you create like a `addTypeToMessages(options)` helper function?